tests/MongoDB.Driver.Tests/Specifications/transactions/TransactionTestRunner.cs

```diff
@@ -79,19 +95,65 @@ public void Run(JsonDrivenTestCase testCase)
         }
 
         // private methods
-        private void Run(BsonDocument shared, BsonDocument test)
+        private void CheckClusterTopology(BsonArray allowedTopologies)
+        {
+            var allowedClusterTypes = allowedTopologies.Select(MapToplogyToClusterType);
+            RequireServer.Check().ClusterTypes(allowedClusterTypes.ToArray());
+
+            var clusterType = CoreTestConfiguration.Cluster.Description.Type;
+            switch (clusterType)
+            {
+                case ClusterType.Sharded:
+                    RequireServer.Check().Supports(Feature.ShardedTransactions);
+                    break;
+                case ClusterType.ReplicaSet:
+                    RequireServer.Check().Supports(Feature.Transactions);
+                    break;
+                default:
+                    throw new Exception("Topology type for transactions must be replicaset or sharded.");
+
+            }
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (!disposing) return;
+            foreach (var disposable in _disposables)
+            {
+                disposable.Dispose();
+            }
```

Comment:
Your code as written is 100% correct, but in all other Dispose methods we have written it like this:

            if (disposing)
            {
                foreach (var disposable in _disposables)
                {
                    disposable.Dispose();
                }
            }